src/app/core/components/input/input.component.ts

```diff
@@ -1,31 +1,61 @@
 import { Component } from '@angular/core';
-import { NgForm } from '@angular/forms';
+import { FormBuilder, FormGroup, NgForm } from '@angular/forms';
 
 import { UserInputService } from '@services/user-input.service';
+import { DataService } from '@core/services/data.service';
 import { YoutubeService } from '@services/youtube.service';
 import { VimeoService } from '@services/vimeo.service';
 import { VideoPlatform } from '@shared/video-platform.model';
+import { ExtrnalErrorStateMatcher } from './external-error-state-matcher';
+import { Content } from '@shared/content.model';
 
 @Component({
   selector: 'app-input',
   templateUrl: './input.component.html',
   styleUrls: ['./input.component.scss'],
 })
 export class InputComponent {
-  constructor(private youtube: YoutubeService, private userInput: UserInputService, private vimeo: VimeoService) {}
+  protected errorMessage = Content.errorDefault;
+  public inputForm!: FormGroup;
+  public externalErrorStateMatcher = new ExtrnalErrorStateMatcher();
 
-  public onAddVideo(form: NgForm): void {
-    if (this.userInput.validatePath(form.form.value.video)) {
+  constructor(
+    private formBuilder: FormBuilder,
+    private userInput: UserInputService,
+    private data: DataService,
+    private youtube: YoutubeService,
+    private vimeo: VimeoService
+  ) {}
+
+  ngOnInit(): void {
+    this.inputForm = this.formBuilder.group({
+      video: [],
+    });
+  }
+
+  public onAddVideo(): void {
+    this.externalErrorStateMatcher.setErrorStateFalse();
+
+    if (this.userInput.validatePath(this.inputForm.value.video)) {
       const dataToFetch = {
-        platform: this.userInput.extractPlatform(form.form.value.video),
-        id: this.userInput.extractId(form.form.value.video),
+        platform: this.userInput.extractPlatform(this.inputForm.value.video),
+        videoId: this.userInput.extractId(this.inputForm.value.video),
       };
-      if (dataToFetch.platform === VideoPlatform.youtube) {
-        this.youtube.fetchVideo(`${dataToFetch.id}`);
+      if (!this.data.checkVideoIn(dataToFetch.videoId)){
+        this.errorMessage = Content.errorVideoExist;
+        this.externalErrorStateMatcher.setErrorStateTrue()
```

Comment:
wydaje mi się że ten externalErrorStateMatcher to jest overkill tutaj i niczego nam nie daje, proponuję inne rozwiązanie
utworzenie dwóch metod validateUrl() validateDuplicateVideo() i w tym pierwszym sprawdzasz czy url lub podane id jest poprawne a w drugim możesz przekazać argument, czyli listę video i sprawdzić czy takie video już jest dodane 
tutaj masz przykład jak można dobić się do custom validatora 
https://blog.angular-university.io/angular-custom-validators/

Reply:
Rzecz w tym, że nie chciałem duplikować funkcji - chciałem użyć metod z serwisu gdzie już robię walidację.